master/cmd/determined-master/root.go

```diff
@@ -79,13 +79,26 @@ func initializeConfig() error {
 	if err != nil {
 		return err
 	}
-	if err = mergeConfigBytesIntoViper(bs); err != nil {
-		return err
+
+	// Write a configMap from the config file, and create a copy (cpMap) to
+	// deepcopyq values needed to override viper's merge auto-lowercasing.
+	var configMap, cpMap map[string]interface{}
+	configErr := yaml.Unmarshal(bs, &configMap)
+	cpErr := yaml.Unmarshal(bs, &cpMap)
+	if configErr != nil || cpErr != nil {
+		return errors.Wrap(err, "error unmarshal yaml configuration file")
+	}
+
+	if err := v.MergeConfigMap(configMap); err != nil {
+		return errors.Wrap(err, "error merge configuration to viper")
 	}
 
 	// Now call viper.AllSettings() again to get the full config, containing all values from CLI flags,
-	// environment variables, and the configuration file.
-	conf, err := getConfig(v.AllSettings())
+	// environment variables, and the configuration file. Override the task_container_defaults value
+	// using the map you copied.
+	viperConfig := v.AllSettings()
+	viperConfig["task_container_defaults"] = cpMap["task_container_defaults"]
```

Comment:
I think we also want to do ```"resource_pools"``` since pools also have a ```task_container_defaults``` under them

https://docs.determined.ai/latest/reference/deploy/config/master-config-reference.html#id13